components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/processors/SPInitLogoutRequestProcessor.java

```diff
@@ -165,6 +171,24 @@ public SAMLSSOReqValidationResponseDTO process(LogoutRequest logoutRequest, Stri
         }
     }
 
+    private void setX509Certificate(String issuer, SAMLSSOServiceProviderDO logoutReqIssuer) {
+
+        try {
+            String tenant = logoutReqIssuer.getTenantDomain();
+            ServiceProvider serviceProvider = SAMLSSOUtil.getApplicationMgtService().
+                    getServiceProviderByClientId(issuer, SAMLSSOConstants.INBOUND_AUTH_TYPE_SAML, tenant);
+            String cert = serviceProvider.getCertificateContent();
+            X509Certificate certificate = (X509Certificate) IdentityUtil.convertPEMEncodedContentToCertificate(cert);
+            logoutReqIssuer.setX509Certificate(certificate);
+
+        } catch (IdentityApplicationManagementException | CertificateException e) {
+            String errorMessage = String.format("An error occurred while retrieving the application " +
+                    "certificate for file based SAML service provider with the issuer name '%s'. " +
+                    "The service provider will NOT be loaded.", issuer);
+            log.error(errorMessage, e);
```

Comment:
Can we continue without throwing the error? If so please indicate that using a small comment